### PR TITLE
add covariance_estimator

### PR DESCRIPTION
**Pull request by matthieugomez**
Return the `CovarianceEstimator` used to compute the variance-covariance matrix

### CODE REVIEW

**Comment by nalimilan** [comment]
Maybe we should just add a `CovarianceEstimator` method? Not sure. Cc: @kleinschmidt

**Comment by kleinschmidt** [comment]
Does statsbase own CovarianceEstimator type?

**Comment by nalimilan** [comment]
Yes: https://github.com/JuliaStats/StatsBase.jl/blob/24a80caf3b2a98adaed24b72e571c482928e15de/src/cov.jl#L158

**Comment by kleinschmidt** [comment]
A few people on slack suggested that the constructor pun would be more appropriate when some kind of computation is involved, which (as far as I can tell) isn't the intended use case here, right?  This would just retrieve an existing `<:CovarianceEstimator`.  So I'm inclined to think that `covariance_estimator` seems more natural.

**Comment by matthieugomez** [comment]
I'm personally indifferent, so let me know.

File: src/statmodels.jl [hunk 1]
```diff
@@ -132,6 +132,13 @@ Return the variance-covariance matrix for the coefficients of the model.
 """
 vcov(obj::StatisticalModel) = error("vcov is not defined for $(typeof(obj)).")
 
+"""
+    covariance_estimator(obj::StatisticalModel)
+
+Return the variance-covariance estimator used to compute the variance-variance matrix
```
Comment (nalimilan) [inline src/statmodels.jl]:
```suggestion
Return the `CovarianceEstimator` used to compute the variance-covariance matrix.
```

**Comment by nalimilan** [comment]
Hard to decide... Another question if we decide to add a function: maybe it should be `vcov_estimator` since it goes with `vcov`?

**Comment by matthieugomez** [comment]
Or use covariance instead of vcov ;)

**Comment by nalimilan** [comment]
Yeah, well, that would be breaking, so... An even more appealing solution would be to overload `cov` from Base, since that's really the same thing. Then you'd call the function `covestimator`, which is a bit shorter.

**Comment by matthieugomez** [comment]
I don't think it's that important actually. Maybe we should close until we have a better story for `covariance` `vcov` etc?

**Comment by nalimilan** [comment]
You mean you don't really need that function? Then yes, maybe better close until we have a use case, that's always better to design good APIs.

**Comment by matthieugomez** [comment]
Exactly

**Pull request closed without merging**